test/k8s_tests/knative_e2e_test.go

```diff
@@ -0,0 +1,273 @@
+		testCapp := mock.CreateBaseCapp()
+		testCapp.Spec.ScaleMetric = "cpu"
+		testCapp.Spec.ConfigurationSpec.Template.Spec.PodSpec.Containers[0].Env = *mock.CreateEnvVarObject(secretName)
+		toBeUpdatedCapp := createAndGetCapp(testCapp)
+
+		By("Checking if the ksvc was created successfully")
+		ksvcObject := mock.CreateKnativeServiceObject(toBeUpdatedCapp.Name)
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, ksvcObject)
+		}, TimeoutCapp, CappCreationInterval).Should(BeTrue(), "Should find a resource.")
+
+		By("Checking if the ksvc has the default annotations")
+		Eventually(func() string {
+			return utilst.GetKsvc(k8sClient, ksvcObject.Name, ksvcObject.Namespace).Annotations[creatorAnnotation]
+		}, TimeoutCapp, CappCreationInterval).Should(Equal(cappServiceAccount))
+		Eventually(func() string {
+			return utilst.GetKsvc(k8sClient, ksvcObject.Name, ksvcObject.Namespace).Annotations[lastModifierAnnotation]
+		}, TimeoutCapp, CappCreationInterval).Should(Equal(cappServiceAccount))
+		updateCappAndCheckRevisionReadiness(toBeUpdatedCapp, 1, true)
+
+		By("Deleting the capp instance")
+		utilst.DeleteCapp(k8sClient, toBeUpdatedCapp)
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, toBeUpdatedCapp)
+		}, TimeoutCapp, CappCreationInterval).ShouldNot(BeTrue(), "Should not find a resource.")
+
+		By("Checking if the ksvc exists")
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, ksvcObject)
+		}, TimeoutCapp, CappCreationInterval).ShouldNot(BeTrue(), "Should not find a resource.")
+
+		By("Checking if the revision exists")
+		revisionObject := mock.CreateRevisionObject(utilst.GetRevisionName(toBeUpdatedCapp.Name, 1))
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, revisionObject)
+		}, TimeoutCapp, CappCreationInterval).ShouldNot(BeTrue(), "Should not find a resource.")
+	})
+
+	It("Should update ksvc metric annotation and create a new revision when updating the capp scale metric", func() {
+		By("Creating a capp instance")
+		testCapp := mock.CreateBaseCapp()
+		testCapp.Spec.ScaleMetric = "cpu"
+		toBeUpdatedCapp := createAndGetCapp(testCapp)
+
+		By("Updating the Capp scale metric")
+		toBeUpdatedCapp.Spec.ScaleMetric = "memory"
+		updateCappAndCheckRevisionReadiness(toBeUpdatedCapp, 2, true)
+
+		By("Checking if the ksvc was updated successfully")
+		Eventually(func() string {
+			ksvc := utilst.GetKsvc(k8sClient, toBeUpdatedCapp.Name, toBeUpdatedCapp.Namespace)
+			return ksvc.Spec.Template.Annotations[knativeMetricAnnotation]
+		}, TimeoutCapp, CappCreationInterval).Should(Equal("memory"))
+	})
+
+	It("Should update ksvc container name and create a new revision when updating a capp container name", func() {
+		By("Creating a capp instance")
+		testCapp := mock.CreateBaseCapp()
+		toBeUpdatedCapp := createAndGetCapp(testCapp)
+
+		By("Updating the a capp container name")
+		newContainerName := utilst.GenerateCappName()
```

Comment:
generate capp name and you creating container ? you can expose a different function for this. for generating random name